hphp/hack/test/typecheck/xhp_modifier/optional_xhp.php

```diff
@@ -0,0 +1,20 @@
+<?hh // partial
+/**
+ * Copyright (c) 2014, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the "hack" directory of this source tree.
+ *
+ *
+ */
+
+class A {
+  public ?:x:base $xhp;
+
+  public function foo(?:x:base $x): ?:x:base {
+    return null;
+  }
+}
+
+xhp class x:base {}
```

Comment:
Most of your tests make sense to me, but I'm not sure what this is testing. Could you add a comment? Something to do with mangling names when they're nullable?